File: src/programmers/level2/카펫.java
package programmers.level2;

public class 카펫 {
    public static void main(String[] args) {
        int brown = 10;
        int yellow = 2;
        solution(brown, yellow);
    }

    public static int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int num = brown + yellow;

        for (int i = 3; i < num; i++) { // 가운데 yellow 이려면 가로세로 3이상이여야 함
            if (num % i == 0 && ((i - 2) * ((num / i) - 2) == yellow)) { // 약수면서
                answer[0] = num / i;
                answer[1] = i;
                break;
            }
        }

        return answer;
    }
}

File: 프로그래머스/lv2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    let num = brown + yellow;
    
    for(let col = 3; col <= num / col; col++){ // 세로
        let row = num / col;
        
        // 한줄로 감싸져 있으므로
        if(((row - 2) * (col - 2)) === yellow)
            return [row, col];
    }
}    